### ISSUE TITLE
[Bug]: CLI Installation Error  on MacOS - Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString stringByStandardizingPath]

### ISSUE THREAD

**Issue by therealhieu** [infinyon/fluvio#3575]
**What happened**
Run:
```
curl -fsS https://hub.infinyon.cloud/install/install.sh | bash 
```

Got this error:
```
fluvio: ⏳ Downloading Fluvio 0.10.15 for x86_64-apple-darwin...
fluvio: ⬇️  Downloaded Fluvio, installing...
fluvio: ✅ Successfully installed /Users/hieunguyen/.fluvio/bin/fluvio-stable
fluvio: ⏳ Downloading Fluvio channel frontend 0.10.15 for x86_64-apple-darwin...
fluvio: ✅ Successfully installed /Users/hieunguyen/.fluvio/bin/fluvio
fluvio: ☁️ Installing Fluvio Cloud...
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString stringByStandardizingPath]: unrecognized selector sent to instance 0x7f86f58055b0'
*** First throw call stack:
(
        0   CoreFoundation                      0x00007ff80d5fa0b6 __exceptionPreprocess + 242
        1   libobjc.A.dylib                     0x00007ff80d0ef231 objc_exception_throw + 48
        2   CoreFoundation                      0x00007ff80d69f3a4 -[NSObject(NSObject) __retain_OA] + 0
        3   CoreFoundation                      0x00007ff80d567833 ___forwarding___ + 1379
        4   CoreFoundation                      0x00007ff80d567248 _CF_forwarding_prep_0 + 120
        5   Foundation                          0x00007ff80e4c811f -[NSProcessInfo arguments] + 203
        6   CoreFoundation                      0x00007ff80d66cce5 __getDefaultArguments_block_invoke + 101
        7   libdispatch.dylib                   0x00007ff80d30459a _dispatch_client_callout + 8
        8   libdispatch.dylib                   0x00007ff80d3057cb _dispatch_once_callout + 20
        9   CoreFoundation                      0x00007ff80d66c6f6 _addBackstopValuesForIdentifierAndSource + 636
        10  CoreFoundation                      0x00007ff80d533a3d __81-[_CFXPreferences(SourceAdditions) withNamedVolatileSourceForIdentifier:perform:]_block_invoke + 151
        11  CoreFoundation                      0x00007ff80d66c3b5 -[_CFXPreferences withNamedVolatileSourceForIdentifier:perform:] + 241
        12  CoreFoundation                      0x00007ff80d53a600 -[CFPrefsSearchListSource addNamedVolatileSourceForIdentifier:] + 98
        13  CoreFoundation                      0x00007ff80d6a6eab __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke.160 + 310
        14  CoreFoundation                      0x00007ff80d6a6b78 -[_CFXPreferences withSearchLists:] + 60
        15  CoreFoundation                      0x00007ff80d535058 __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke + 278
        16  CoreFoundation                      0x00007ff80d6a6cf0 -[_CFXPreferences withSearchListForIdentifier:container:cloudConfigurationURL:perform:] + 349
        17  CoreFoundation                      0x00007ff80d534ae2 -[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:] + 124
        18  CoreFoundation                      0x00007ff80d534a28 _CFPreferencesCopyAppValueWithContainerAndConfiguration + 101
        19  SystemConfiguration                 0x00007ff80e17f8bb SCDynamicStoreCopyProxiesWithOptions + 155
        20  fluvio                              0x000000010f0a81cb Curl_macos_init + 11
        21  fluvio                              0x000000010f0971dd curl_global_init + 141
        22  fluvio                              0x000000010f08460d _ZN3std4sync4once4Once9call_once28_$u7b$$u7b$closure$u7d$$u7d$17h9e5dce304539c317E + 29
        23  fluvio                              0x000000010f3eb068 _ZN3std10sys_common4once5queue4Once4call17hf241db9abbfabd67E + 984
        24  fluvio                              0x000000010f084927 _ZN4curl9INIT_CTOR9init_ctor17h4ce73ebde4459de0E + 71
        25  dyld                                0x00007ff80d13bf16 ___ZZNK5dyld46Loader25findAndRunAllInitializersERNS_12RuntimeStateEENK3$_0clEv_block_invoke + 182
        26  dyld                                0x00007ff80d17d584 ___ZNK5dyld313MachOAnalyzer18forEachInitializerER11DiagnosticsRKNS0_15VMAddrConverterEU13block_pointerFvjEPKv_block_invoke.175 + 133
        27  dyld                                0x00007ff80d171913 ___ZNK5dyld39MachOFile14forEachSectionEU13block_pointerFvRKNS0_11SectionInfoEbRbE_block_invoke + 543
        28  dyld                                0x00007ff80d11f07f _ZNK5dyld39MachOFile18forEachLoadCommandER11DiagnosticsU13block_pointerFvPK12load_commandRbE + 249
        29  dyld                                0x00007ff80d170adc _ZNK5dyld39MachOFile14forEachSectionEU13block_pointerFvRKNS0_11SectionInfoEbRbE + 176
        30  dyld                                0x00007ff80d173104 _ZNK5dyld39MachOFile32forEachInitializerPointerSectionER11DiagnosticsU13block_pointerFvjjRbE + 116
        31  dyld                                0x00007ff80d17d2ba _ZNK5dyld313MachOAnalyzer18forEachInitializerER11DiagnosticsRKNS0_15VMAddrConverterEU13block_pointerFvjEPKv + 390
        32  dyld                                0x00007ff80d138c48 _ZNK5dyld46Loader25findAndRunAllInitializersERNS_12RuntimeStateE + 222
        33  dyld                                0x00007ff80d13e6ab _ZNK5dyld416JustInTimeLoader15runInitializersERNS_12RuntimeStateE + 21
        34  dyld                                0x00007ff80d138e3d _ZNK5dyld46Loader23runInitializersBottomUpERNS_12RuntimeStateERN5dyld35ArrayIPKS0_EE + 181
        35  dyld                                0x00007ff80d13bf8c _ZZNK5dyld46Loader38runInitializersBottomUpPlusUpwardLinksERNS_12RuntimeStateEENK3$_1clEv + 98
        36  dyld                                0x00007ff80d138ed3 _ZNK5dyld46Loader38runInitializersBottomUpPlusUpwardLinksERNS_12RuntimeStateE + 93
        37  dyld                                0x00007ff80d15a9ed _ZN5dyld44APIs25runAllInitializersForMainEv + 317
        38  dyld                                0x00007ff80d124261 _ZN5dyld4L7prepareERNS_4APIsEPKN5dyld313MachOAnalyzerE + 3397
        39  dyld                                0x00007ff80d12333f start + 1839
)
libc++abi: terminating due to uncaught exception of type NSException
bash: line 485: 94708 Abort trap: 6           FLUVIO_BOOTSTRAP=true CHANNEL_BOOTSTRAP=${VERSION} "${FLUVIO_BIN}/fluvio" install fluvio-cloud
```

**Environment (please complete the following information):**
 - OS: MacOS Sonoma 14.0.0 - Intel
 - Fluvio 0.10.15

**Comment by ajhunyady** [comment]
@therealhieu thanks for reporting. We'll take a look.

**Comment by sehz** [comment]
Confirm issue.  Upgraded to Sonoma and seeing problem

**Comment by digikata** [comment]
A possible problem is  with libcurl if it's the same as identified here: https://github.com/rust-lang/cargo/issues/12670

**Comment by sehz** [comment]
Where are we using curl? (downloading)?  Either replace it or bundle in instead of dynamic linking

**Comment by digikata** [comment]
Yes, I have to do some more work to confirm it, but it's likely the http-client we use for download is configured to use the system libcurl

**Comment by digikata** [comment]
Testing a build fix that uses the system curl on macos-sonoma.  The build is normally configured to use a static-curl.

**Comment by sehz** [comment]
Can we add a feature flag to configure `curl` embedding?

**Comment by digikata** [comment]
A feature flag reconfig for macos was the quick fix, I'm taking a little time for deeper fix to remove curl completely as a dep given it might need another update soon.

**Comment by digikata** [comment]
Note: This issue is now corrected w/ the 0.10.16 release. If you are getting an exception on Macos Sonoma with a NSInvalidArgumentException from an existing fluvio install, you may need to reinstall the cli from curl https://www.fluvio.io/docs/get-started/mac/
